### PR TITLE
Problem: Outputs API doesn't respond with correct unspent/spent outputs

### LINKED ISSUE #2552: BigchainDB outputs endpoint is not giving correct unspent outputs
bigchaindb JS driver tests run by Travis are failing.
The js driver is using the master branch of BigchainDB.
How to reproduce:

-a) Run this test: https://github.com/bigchaindb/js-bigchaindb-driver/blob/master/test/integration/test_integration.js#L286

-b) Create and post a CREATE transaction with three different outputs with the same public key. Create and post a TRANSFER transaction that spends the second output of the previous transaction and send it to another different public key. Query for spend outputs for the first public key. You won't find any output spent. 

-c) Code snippet to reproduce the error for node.js:
```js
var BigchainDB = require('bigchaindb-driver')
var bip39 = require('bip39')

const alice = new BigchainDB.Ed25519Keypair(bip39.mnemonicToSeed('alice one').slice(0, 32))
const bob = new BigchainDB.Ed25519Keypair(bip39.mnemonicToSeed('bob one').slice(0, 32))

const API_PATH = 'http://127.0.0.1:9984/api/v1/'
const conn = new BigchainDB.Connection(API_PATH, {
     app_id: '',
    app_key: ''
})


const aliceOutput = BigchainDB.Transaction.makeOutput(
    BigchainDB.Transaction.makeEd25519Condition(alice.publicKey))
const bobOutput = BigchainDB.Transaction.makeOutput(
        BigchainDB.Transaction.makeEd25519Condition(bob.publicKey))

Create()
async function Create(){
    const txCreate = BigchainDB.Transaction.makeCreateTransaction(
        // Define the asset to store, in this example it is the current temperature
        // (in Celsius) for the city of Berlin.
        {
            'timestamp': Date.now()
        }, {
            'entity': 'aaa'
        },
        // enemy is the owner
        [aliceOutput, aliceOutput, aliceOutput],
        alice.publicKey
    )
    
    // Sign the transaction with private keys
    const createSigned = BigchainDB.Transaction.signTransaction(txCreate, alice.privateKey)
    
    await conn.postTransactionCommit(createSigned)
        .then(res => {
            console.log('createEnemy function finished. ', res.id, 'accepted')
        })
    
    
    const createTranfer = BigchainDB.Transaction.makeTransferTransaction(
        [{
            tx: createSigned,
            output_index: 1
        }],
        [bobOutput], {
            'timestamp': Date.now()
        }
    
    )
    const signedTransfer = BigchainDB.Transaction.signTransaction(createTranfer, alice.privateKey)
    
    await conn.postTransactionCommit(signedTransfer)
        .then(res => {
            console.log('updateGroup function finished, ', res.id, 'accepted')
        })
    

    const spentOutputs = await conn.listOutputs(alice.publicKey, true)
    console.log('Spent outputs, expected 1', spentOutputs)
    const unspenttOutputs = await conn.listOutputs(alice.publicKey, false)
    console.log('Spent outputs, expected 2', unspenttOutputs)
}

```

### LINKED ISSUE #2568: BigchainDB lets transfer an asset after it is transferred to another owner
I create an asset with Alice as owner and then i make a transaction to Bobr but BigchainDB lets me make another transaction from Alice to Peter.

**Ubuntu 16.04
BigchainDB 2.0.0b6**

**CODE:** (JavaScript)

```javascript
const driver = require('bigchaindb-driver')
const API_PATH = 'http://localhost:9984/api/v1/'
const conn = new driver.Connection(API_PATH)

const alice = new driver.Ed25519Keypair()
const bob = new driver.Ed25519Keypair()
const peter = new driver.Ed25519Keypair()

console.log('Alice: ', alice.publicKey)
console.log('Bob: ', bob.publicKey)
console.log('Peter: ', peter.publicKey)

const assetdata = {
    'bicycle': {
            'serial_number': 'abcd1234',
            'manufacturer': 'Bicycle Inc.',
    }
}

const metadata = {'planet': 'earth'}

const txCreateAliceSimple = driver.Transaction.makeCreateTransaction(
    assetdata,
    metadata,

    // A transaction needs an output
    [ driver.Transaction.makeOutput(
            driver.Transaction.makeEd25519Condition(alice.publicKey))
    ],
    alice.publicKey
)

const txCreateAliceSimpleSigned = driver.Transaction.signTransaction(txCreateAliceSimple, alice.privateKey)

conn.postTransactionCommit(txCreateAliceSimpleSigned)
    .then(retrievedTx => {
        console.log('Transaction', retrievedTx.id, 'successfully posted.')
    })
    .then(() => {
        const txTransferBob = driver.Transaction.makeTransferTransaction(
            // signedTx to transfer and output index
            [{ tx: txCreateAliceSimpleSigned, output_index: 0 }],
            [driver.Transaction.makeOutput(driver.Transaction.makeEd25519Condition(bob.publicKey))],
            // metadata
            {price: '100 euro'}
        )

        let txTransferBobSigned = driver.Transaction.signTransaction(txTransferBob, alice.privateKey);

        return conn.postTransactionCommit(txTransferBobSigned)
    })
    .then(tx => {
        console.log('Response from BDB server:', tx)
        console.log('Is Bob the owner?', tx['outputs'][0]['public_keys'][0] == bob.publicKey)
        console.log('Was Alice the previous owner?', tx['inputs'][0]['owners_before'][0] == alice.publicKey )
    })
    .then(() => {
        console.log("Another transaction from Alice, but now she is not the owner")
        const txTransferPeter = driver.Transaction.makeTransferTransaction(
            // signedTx to transfer and output index
            [{ tx: txCreateAliceSimpleSigned, output_index: 0 }],
            [driver.Transaction.makeOutput(driver.Transaction.makeEd25519Condition(peter.publicKey))],
            // metadata
            {price: '200 euro'}
        )

        let txTransferPeterSigned = driver.Transaction.signTransaction(txTransferPeter, alice.privateKey);

        return conn.postTransactionCommit(txTransferPeterSigned)
    })
    .then(tx => {
        console.log('Response from BDB server:', tx)
        console.log('Is Bob the owner?', tx['outputs'][0]['public_keys'][0] == bob.publicKey)
        console.log('Is Peter the owner?', tx['outputs'][0]['public_keys'][0] == peter.publicKey)
        console.log('Was Alice the previous owner?', tx['inputs'][0]['owners_before'][0] == alice.publicKey )       
    })
    .then(() => {
        console.log("FININSH")
    })
```

**TERMINAL:**
```bash
Alice:  GYmQqKp561uRyQqH54whLwbVCS9PzN8R6372AZZqXVqi
Bob:  j9bUqPELQQCrpjeVhodZLrDHdf4R8XrvrMdEHC9Zxfc
Peter:  6UPWcgQKhGgbmZvmnz7CNgA7Qd9tofWBSp3kMFt9z8Pu
Transaction 322ba8f35688cb2142314f3cb4701784455bdfd0a90cba916ac597106755d551 successfully posted.
Response from BDB server: { asset:
   { id:
      '322ba8f35688cb2142314f3cb4701784455bdfd0a90cba916ac597106755d551' },
  id:
   '90093f67e630f582c2c0dd8eaba6550c1d949f84bb6fe2aed2167a4c2a0fb6af',
  inputs:
   [ { fulfillment:
        'pGSAIOcChkmXLDlyYDH3cxTRtsZeTEpjIk3Nw9OowQ_dYeYpgUBlkDe-16g7mM47TaNR5DP3p3FFg3F5TQKvlDhpRHd6PYl5-nc88SZumNupCn6niJgujyjP4WFX_ebmO2eSsDEC',
       fulfills: [Object],
       owners_before: [Array] } ],
  metadata: { price: '100 euro' },
  operation: 'TRANSFER',
  outputs:
   [ { amount: '1', condition: [Object], public_keys: [Array] } ],
  version: '2.0' }
Is Bob the owner? true
Was Alice the previous owner? true
Another transaction from Alice, but now she is not the owner
Response from BDB server: { asset:
   { id:
      '322ba8f35688cb2142314f3cb4701784455bdfd0a90cba916ac597106755d551' },
  id:
   'b78d0d26d39477e6ed9a44f8d97b999f3f2a2e7303aea6fdfb6c154a52f2f65b',
  inputs:
   [ { fulfillment:
        'pGSAIOcChkmXLDlyYDH3cxTRtsZeTEpjIk3Nw9OowQ_dYeYpgUDmPm5CYHwSryHHBGENUpY1bnN5DzpJ0tlhsS5lCfu2Fk2vACYLe4Co_Nzo6JsetQTXos29ye_zqhwkGE7XIaEC',
       fulfills: [Object],
       owners_before: [Array] } ],
  metadata: { price: '200 euro' },
  operation: 'TRANSFER',
  outputs:
   [ { amount: '1', condition: [Object], public_keys: [Array] } ],
  version: '2.0' }
Is Bob the owner? false
Is Peter the owner? true
Was Alice the previous owner? true
FININSH
```

### PR DESCRIPTION
**Pull request by kansi**
Solution: Fix `fastquery` such that embedded document is queried properly

Fixes #2552 
Fixes #2568

### CODE REVIEW

File: bigchaindb/common/transaction.py [hunk 1]
```diff
@@ -274,6 +274,15 @@ def to_dict(self):
                 'output_index': self.output,
             }
 
+    def to_query(self):
+        if self.txid is None and self.output is None:
```
Comment (ttmc) [inline bigchaindb/common/transaction.py]: Is it possible for just one of these to be `None`?

Comment (kansi) [reply bigchaindb/common/transaction.py]: Actually, these fields cannot be `None`. I started off with `to_dict`, this code is leftover

File: bigchaindb/common/transaction.py [hunk 2]
```diff
@@ -274,6 +274,12 @@ def to_dict(self):
                 'output_index': self.output,
             }
 
+    def to_query(self):
```
[the remark this answers is not in the document]
Comment (ldmberman) [reply bigchaindb/common/transaction.py]: Actually, `$and` is applied to `fulfills` so:
```
{
    'inputs.fulfills: {
        '$and': [{
            transaction_id': {
                '$in': transaction_ids,
            },
            'output_index': {
                '$in': output_indexes,
            },
       }
    }
]
```

[the remark this answers is not in the document]
Comment (ldmberman) [reply bigchaindb/common/transaction.py]: Ah, and we also seem to need `$elemMatch`.

**Pull request merged**